src/main/java/nl/tudelft/contextproject/model/entities/VRPlayer.java

```diff
@@ -225,13 +219,20 @@ public void pickUp() {
 		Set<Entity> set = Main.getInstance().getCurrentGame().getEntities();
 
 		for (Entity ent : set) {
+
 			if (!ent.collidesWithPlayer(INTERACT_RANGE)) continue;
 
 			if (ent instanceof Bomb) {
-				inventory.add(new Bomb());
-				ent.setState(EntityState.DEAD);
-				return;
-			} else if (ent instanceof Key) {
+				if (inventory.numberOfBombs() < 1) {
+					if (!((Bomb) ent).getPickedup()) {
```

Comment:
These 3 ifs can be combined, the if (ent instanceof Key) can be changed back to an else if.

So `if (ent instanceof Bomb && inventory.numberOfBombs() < 1 && !((Bomb ent).getPickedup()) { ... }`